pkg/reconcilers/shared/postcreate_hook.go

```diff
@@ -45,72 +45,82 @@ func (r *BaseReconciler) ReconcileUpdatePostCreateHook(ctx context.Context, hcp
     logger := clog.FromContext(ctx)
     namespace := util.GenerateNamespaceFromControlPlaneName(hcp.Name)
     
-    if hcp.Spec.PostCreateHook == nil {
-        return nil
-    }
-
-    // Hook is only applied once after CP creation
-    _, ok := hcp.Status.PostCreateHooks[*hcp.Spec.PostCreateHook]
-    if ok {
-        return nil
-    }
-
-    logger.Info("Running ReconcileUpdatePostCreateHook", "post-create-hook", *hcp.Spec.PostCreateHook)
-
-    // Get the post create hook
-    hook := &v1alpha1.PostCreateHook{
-        ObjectMeta: metav1.ObjectMeta{
-            Name: *hcp.Spec.PostCreateHook,
-        },
-    }
-    if err := r.Client.Get(context.TODO(), client.ObjectKeyFromObject(hook), hook, &client.GetOptions{}); err != nil {
-        return fmt.Errorf("error retrieving post create hook %s: %w", *hcp.Spec.PostCreateHook, err)
-    }
-
-    // Build variables with proper precedence:
-    // 1. Default vars from PostCreateHook
-    // 2. User-provided vars from ControlPlane (override defaults)
-    // 3. Built-in system vars (highest priority)
-    vars := make(map[string]interface{})
+    // Collect all hooks to process (legacy + new)
+    var hooks []v1alpha1.PostCreateHookUse
     
-    // Add default variables from hook spec
-    for _, dv := range hook.Spec.DefaultVars {
-        vars[dv.Name] = dv.Value
+    // Add legacy hook if specified
+    if hcp.Spec.PostCreateHook != nil {
+        hooks = append(hooks, v1alpha1.PostCreateHookUse{
+            HookName: hcp.Spec.PostCreateHook,
+            Vars:     hcp.Spec.PostCreateHookVars,
+        })
```

Comment:
If the same hook is specified both as the legacy field and in the new array, it may be applied twice unless handled elsewhere.  Add a check to avoid duplicate execution of hooks (e.g., by hook name).